src/cpl/share_esmf/cropcalStreamMod.F90

```diff
@@ -613,6 +736,90 @@ subroutine cropcal_interp(bounds, num_pcropp, filter_pcropp, init, crop_inst)
+             fldptr1=dataptr1d_gdd20_season_start,  rc=rc)
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) then
+           call ESMF_Finalize(endflag=ESMF_END_ABORT)
+        end if
+        call dshr_fldbun_getFldPtr(sdat_cropcal_gdd20_season_end%pstrm(1)%fldbun_model, trim(stream_varnames_gdd20_season_enddate(n)), &
+             fldptr1=dataptr1d_gdd20_season_end,  rc=rc)
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) then
+           call ESMF_Finalize(endflag=ESMF_END_ABORT)
+        end if
+        ! Note that the size of dataptr1d includes ocean points so it will be around 3x larger than lsize
+        ! So an explicit loop is required here
+        do g = 1,lsize
+
+           ! If read-in value is invalid, set to -1. Will be handled later in this subroutine.
+           if (dataptr1d_gdd20_season_start(g) <= 0 .or. dataptr1d_gdd20_season_start(g) > 366 &
+               .or. dataptr1d_gdd20_season_start(g) /= dataptr1d_gdd20_season_start(g)) then
+              dataptr1d_gdd20_season_start(g) = -1
+           end if
+           if (dataptr1d_gdd20_season_end(g) <= 0 .or. dataptr1d_gdd20_season_end(g) > 366 &
+               .or. dataptr1d_gdd20_season_end(g) /= dataptr1d_gdd20_season_end(g)) then
+              dataptr1d_gdd20_season_end  (g) = -1
+           end if
+
+          dataptr2d_gdd20_season_start(g,n) = dataptr1d_gdd20_season_start(g)
+          dataptr2d_gdd20_season_end  (g,n) = dataptr1d_gdd20_season_end  (g)
+        end do
+     end do
+
+     ! Set gdd20 season for each gridcell/patch combination
+     do fp = 1, num_pcropp
+        p = filter_pcropp(fp)
+        ivt = patch%itype(p)
+        ! Will skip generic crops
+        if (ivt >= npcropmin) then
+           n = ivt - npcropmin + 1
+           ! vegetated pft
+           ig = g_to_ig(patch%gridcell(p))
+
+           gdd20_season_starts(p) = real(dataptr2d_gdd20_season_start(ig,n), r8)
+           gdd20_season_ends(p)   = real(dataptr2d_gdd20_season_end  (ig,n), r8)
+       else
+           write(iulog,'(a,i0)') 'cropcal_interp(), gdd20 seasons: Crop patch has ivt ',ivt
+           call ESMF_Finalize(endflag=ESMF_END_ABORT)
+        endif
+     end do
+
+     ! Handle invalid gdd20 season values
+     if (any(gdd20_season_starts(begp:endp) < 1._r8 .or. gdd20_season_ends(begp:endp) < 1._r8)) then
+         ! Fail if not allowing fallback to paramfile sowing windows
+         if ((.not. allow_invalid_gdd20_season_inputs) .and. any(gdd20_season_starts(begp:endp) < 1._r8 .and. patch%wtgcell(begp:endp) > 0._r8 .and. patch%itype(begp:endp) >= npcropmin)) then
+             write(iulog, *) 'At least one crop in one gridcell has invalid gdd20 season start date(s). To ignore and fall back to paramfile sowing windows, set allow_invalid_gdd20_season_inputs to .true.'
```

Comment:
Should say start **or end**.

Reply:
"To ignore and fall back to paramfile sowing windows **for such crop-gridcells**"

Reply:
Done with 592476f7c.